cypress/cypress/integration/mentor-card.spec.ts

```diff
@@ -0,0 +1,168 @@
+    cy.get("[data-cy=my-mentor-card]").contains(
+      "This Mentor can't be built yet."
+    );
+    cy.get("[data-cy=stage-progress]").should("exist");
+    cy.get("[data-cy=my-mentor-card]")
+      .find("[data-cy=next-stage-info]")
+      .trigger("mouseover");
+    cy.contains("This Mentor can select questions from a list");
+    cy.get("[data-cy=thumbnail-wrapper]").trigger("mouseover");
+    cy.get("[data-cy=upload-file]").should("exist");
+    cy.get("[data-cy=bash-button]").should("exist");
+  });
+
+  it("shows placeholder when no thumbnail", () => {
+    cySetup(cy);
+    const testClint = {
+      ...clint,
+      thumbnail: "",
+    };
+    cyMockDefault(cy, { mentor: testClint });
+    cy.visit("/");
+    cy.get("[data-cy=placeholder-thumbnail]").should("exist");
+  });
+
+  it("switches to new image when uploaded", () => {
+    cySetup(cy);
+    const testClint = {
+      ...clint,
+      thumbnail: "url",
+    };
+    cyMockDefault(cy, { mentor: testClint });
+    cy.visit("/");
+
+    cy.get("[data-cy=uploaded-thumbnail]").should("exist");
+  });
+
+  it("does not show toast on incomplete level", () => {
+    cySetup(cy);
+    cyMockDefault(cy, { mentor: clint });
+    cy.visit("/");
+    cy.get("[data-cy=stage-toast]").should("not.exist");
+  });
+
+  it("shows mentor scope toast on stage floor", () => {
+    cySetup(cy);
+    const testClint = {
+      ...clint,
+      answers: clint.answers.map((a, i) => {
+        return i !== 4
+          ? a
+          : {
+              ...a,
+              status: Status.COMPLETE,
+            };
+      }),
+    };
+    cyMockDefault(cy, { mentor: testClint });
+    cy.visit("/");
+    cy.get("[data-cy=stage-toast]").contains(
+      "Your mentor has reached the Scripted stage!"
+    );
+    cy.get("[data-cy=stage-toast]").contains("You have 5 total questions.");
+  });
+
+  it("does not show progress bar at maxed level", () => {
+    cySetup(cy);
+    const testClint = {
+      ...clint,
+      answers: [
+        ...Array(1000)
+          .fill(0)
+          .map((x) => ({
+            _id: "A1_1_1",
+            question: {
+              _id: "A1_1_1",
+              question: "Who are you and what do you do?",
+              type: QuestionType.QUESTION,
+              name: null,
+              paraphrases: [],
+            },
+            transcript:
+              "My name is Clint Anderson and I'm a Nuclear Electrician's Mate",
+            status: Status.COMPLETE,
+          })),
+      ],
+    };
+    cyMockDefault(cy, { mentor: testClint });
+    cy.visit("/");
+    cy.get("[data-cy=my-mentor-card]").contains("Scope: Life-Story");
+    cy.get("[data-cy=stage-progress]").should("not.exist");
+  });
+});
+describe("Bash Button Guides User To Improve Mentor", () => {
+  const bashClint = {
+    ...clint,
+  };
+
+  it("Asks user without thumbnail to upload one", () => {
+    cySetup(cy);
+    bashClint.thumbnail = "";
```

Comment:
never edit anything scoped outside of a test function inside a test function. That is a recipe for race-condition errors and broken isolation

```typescript
cyMockDefault(cy, { 
  mentor: {
    ...bashClint,
    thumbnail: ""
  }
});